xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
> If you don't want my commit, plan B is to remove all uses of

Nope. Only this particular method does not require locking. See your commit message. It's only VideoPlayer thread that can change the vector. The short term solution is not calling this method from outside.